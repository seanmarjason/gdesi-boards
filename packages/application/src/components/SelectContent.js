'use client'

import * as React from 'react';
import NextLink from 'next/link';

import Avatar from '@mui/material/Avatar';
import MuiListItemAvatar from '@mui/material/ListItemAvatar';
import MenuItem from '@mui/material/MenuItem';
import ListItemText from '@mui/material/ListItemText';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListSubheader from '@mui/material/ListSubheader';
import Select, { selectClasses } from '@mui/material/Select';
import Divider from '@mui/material/Divider';
import AddRoundedIcon from '@mui/icons-material/AddRounded';
import FilterNoneIcon from '@mui/icons-material/FilterNone';
import DevicesRoundedIcon from '@mui/icons-material/DevicesRounded';
import SettingsIcon from '@mui/icons-material/Settings';
import ConstructionRoundedIcon from '@mui/icons-material/ConstructionRounded';


const ListItemAvatar = MuiListItemAvatar

export default function SelectContent({ boardId=null}) {
  const [company, setCompany] = React.useState('');

  const handleChange = (event) => {
    setCompany(event.target.value);
  };

  return (
    <Select
      labelId="company-select"
      id="company-simple-select"
      value={company}
      onChange={handleChange}
      displayEmpty
      inputProps={{ 'aria-label': 'Select company' }}
      fullWidth
      sx={{
        maxHeight: 56,
        width: 215,
        '&.MuiList-root': {
          p: '8px',
        },
        [`& .${selectClasses.select}`]: {
          display: 'flex',
          alignItems: 'center',
          gap: '2px',
          pl: 1,
        },
      }}
    >
      <MenuItem value="">
        <ListItemAvatar>
          <Avatar alt={boardId}>
            <DevicesRoundedIcon sx={{ fontSize: '1rem' }} />
          </Avatar>
        </ListItemAvatar>
        <ListItemText primary={boardId} />
      </MenuItem>
      <Divider sx={{ mx: -1 }} />

      { boardId && 
        <MenuItem value={40} component={NextLink} href={`/boards/${boardId}/settings`}>
          <ListItemIcon>
            <SettingsIcon />
          </ListItemIcon>
          <ListItemText primary="Board Settings" />
        </MenuItem>
      }

      <MenuItem value={40} component={NextLink} href="/boards">
        <ListItemIcon>
          <FilterNoneIcon />
        </ListItemIcon>
        <ListItemText primary="Switch Boards" />
      </MenuItem>

      <MenuItem value={40} component={NextLink} href="/boards/new">
        <ListItemIcon>
          <AddRoundedIcon />
        </ListItemIcon>
        <ListItemText primary="Add a board" />
      </MenuItem>
    </Select>
  );
}
