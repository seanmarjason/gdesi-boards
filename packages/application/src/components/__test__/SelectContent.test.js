import { render } from '@testing-library/react'
import SelectContent from '../SelectContent';

it('SelectContent renders correctly', () => {
  const { container } = render(<SelectContent />)
  expect(container).toMatchSnapshot()
})
