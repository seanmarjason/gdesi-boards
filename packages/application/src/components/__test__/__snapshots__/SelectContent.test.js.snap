// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectContent renders correctly 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiSelect-root css-19uj6ei-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Select company"
      aria-labelledby="company-select company-simple-select"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="company-simple-select"
      role="combobox"
      tabindex="0"
    >
      <div
        class="MuiListItemAvatar-root css-1ulznza-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-18k2bs-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wdc28j-MuiSvgIcon-root"
            data-testid="DevicesRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 7c0-.55.45-1 1-1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-1.1 0-2 .9-2 2v11h-.5c-.83 0-1.5.67-1.5 1.5S.67 20 1.5 20H14v-3H4zm19 1h-6c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1m-1 9h-4v-7h4z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
      />
    </div>
    <input
      aria-hidden="true"
      aria-invalid="false"
      class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
      tabindex="-1"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-w4cd9x"
      >
        <span
          aria-hidden="true"
          class="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
